### PR TITLE
USWDS theme for keycloak

### PR DESCRIPTION
**Pull request by awolfe76**
_The number of files (119) is due to the files in USWDS. We can probably clean it up later, but it doesn't hurt anything._

- adds in [uswds](https://standards.usa.gov/) files
- customizes login
hmda-login
- customizes registration
hmda-register
- customizes password reset
hmda-reset-pwd
- custom styles for errors
hmda-errors

### CODE REVIEW

**Comment by hkeeler** [comment]
@awolfe76, this is awesome!  Taking a crack at getting this all running locally now.

File: keycloak/themes/hmda/login/register.ftl [hunk 1]
```diff
@@ -155,43 +110,22 @@
...
-                           '               <i class="fa fa-gavel" aria-hidden="true"></i> ' + institution.regulator +
-                           '            </div>' +
-                           '            <div class="col-md-8">' +
-                           '               <i class="fa fa-envelope" aria-hidden="true"></i> ' + institution.domain[0] +
-                           '            </div>' +
-                           '          </div>' +
-
+                    return '<div class="usa-grid-full">' +
+                           //'    <div styles="vertical-align:middle;" class="col-md-1">' +
+                           //'      <h1>' + getStatusIcon(getFormEmail(), institution.domain[0]) + '</h1>' +
```
Comment (hkeeler) [inline keycloak/themes/hmda/login/register.ftl]: Is there a plan to re-incorporate some form of status indicator?  Without this, there's no way for the user to know ahead of time if the institution they've selected is going to work.

I realize there's a bit less real estate now, so this is likely a question of _where do we put it?_

**Comment by hkeeler** [comment]
@awolfe76, just curious, how much did you have to change the templates themselves?  Are these copied directly, and just the CSS classes updated, or was there more too it.  I'm asking primarily to better understand how much work it is going to be with each upgrade, and how we're going to have to re-check the templates for necessary changes each time.

One thought, it may be more upgrade-friendly to keep keycloak variables for the CSS class names, and update the values in the `theme.properties` file(s) as they do in there templates.

Thoughts?

File: keycloak/themes/hmda/login/theme.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 parent=keycloak
 scripts=lib/jquery/jquery-1.10.2.js lib/select2-3.4.1/select2.min.js
-styles=lib/patternfly/css/patternfly.css lib/zocial/zocial.css lib/select2-3.4.1/select2.css css/login.css
+styles=css/uswds.min.css lib/select2-3.4.1/select2.css css/select2-custom.css
 
-institutionSearchUri={{INSTITUTION_SEARCH_URI}}
```
Comment (awolfe76) [inline keycloak/themes/hmda/login/theme.properties]: Yeah, sorry. Didn't commit that before but must have slipped through at some point.

File: keycloak/themes/hmda/login/register.ftl (hunk 1 above)
Comment (awolfe76) [inline keycloak/themes/hmda/login/register.ftl]: Yes. But I think it's ok for now, we'll figure it out. That's why I left the reminder there. 😄 
I hope that the meeting today will tell us a bit more.

**Comment by awolfe76** [comment]
@hkeeler 
> how much did you have to change the templates themselves? 

It was all of the above, copied, modified CSS classes, and modified the HTML structure too (mostly due to the new CSS classes).

> I'm asking primarily to better understand how much work it is going to be with each upgrade, and how we're going to have to re-check the templates for necessary changes each time.

I'm not worried about the upgrade path on the templates at all, even without using `theme.properties` to store the classes. It seems like they did it that way so others could easily import their themes. Unless we decide to make our own base, can't see why we would, or they change their theme engine, totally different story, we'll be fine.

**Comment by awolfe76** [comment]
@hkeeler - when looking at the files after the conversation here I thought it would be good to clean them up a bit, see the last 2 commits.

To view without whitespace changes:
https://github.com/cfpb/hmda-platform-auth/pull/31/commits/d20c723b3f40054f6ed588ad4cf1a08542975c6a?w=1
https://github.com/cfpb/hmda-platform-auth/pull/31/commits/d1252cf196aeded03c592ce1b17ea50fee39d505?w=1

File: keycloak/themes/hmda/login/register.ftl (hunk 1 above)
Comment (hkeeler) [inline keycloak/themes/hmda/login/register.ftl]: Cool by me.  I created https://github.com/cfpb/hmda-platform-auth/issues/32 so we don't forget about it.

**Comment by hkeeler** [comment]
@awolfe76, thanks for the cleanup.  This looks good to me.  I'd like @wpears to weigh in if possible before merging.

Also, I'm working on a _How to Upgrade Keycloak_ doc.  I'll reach out when I get to the bit about templates.

**Pull request merged**